Make sure hidden fields are not being posted, when they are in a hidden piece of a dialog.

rubnig opened the pull request:
With this feature, we will disable hidden input fields that are part of the _hidden_ parts of a dialog.

This ensures that we only post those fields to the repository when they are part of a visible part of the dialog. If we don't to this and two hidden fields have the same name and a different value, they will be both stored in a single field as an array property with two values.

Commit:
Make sure hidden fields are not being posted, when they are in a hidden piece of a dialog.

stefanseifert commented:
thanks for the PR!

i updated our sample app to test this fix: https://github.com/wcm-io/io.wcm.wcm.ui.granite.sample-app/pull/16

it fixes the problem with the hidden fields, only the value of the hidden field of the visible section is saved.

however, in my test it lead to unwanted severe side effects: all other fields in the shown dialog section are now set to disabled as well when switching a bit between the options. rendering the whole dialog unusable. please cross-check!

example screenshot of the problem:
<img width="400" alt="image" src="https://github.com/user-attachments/assets/1e2bf1d6-95a0-4173-b9ca-545baa4ac463" />

Commit:
Fix hidden input for AEMaaCS dialog. The coral elements render hidden inputs for AEMaaCS + respect delete hints for hidden fields

rubnig commented:
My bad, I didn't test this on AEMaaCS, only on on-prem installations. Apparently the coral ui elements in AEMaaCS render differently then on-premise installations do.

I've just fixed it in: https://github.com/wcm-io/io.wcm.wcm.ui.granite/pull/59/commits/83eed6a716b5d14d11aebe80b8ee73a8d4e2aec4

Nice to see that there is also a sample app available, did completely miss this, but I really like this approach. Also updated the sample app a little bit, so it will always store the hidden input, based on the selected option. See: https://github.com/wcm-io/io.wcm.wcm.ui.granite.sample-app/pull/18

stefanseifert committed:
fix indentation

stefanseifert commented:
lgtm! - i did a quick test in AEMaaCS SDK, AEM 6.5 and AEM 6.6 and it looks fine now

henrykuijpers commented:
FYI, this causes #67 .
